### PR TITLE
generic: fix silicon labs spidev bindings

### PR DESCRIPTION
**Pull request by vtremblay**
Fix Silicon Labs bindings in the spidev driver

Some bindings for Silicon Labs chips already exists upstream. These bindings can be found in trivial-devices.yaml.

The existing bindings are using "silabs" instead of "siliconlabs" to identify the manufacturer.

This commit add two submitted patches for silabs chips and rename the manufacturer in the different DTS for more coherence.

The patches also fix the following message log during the boot of some devices:

```
SPI driver spidev has no spi_device_id for siliconlabs,si3210
```

## References
The full discussion about this PR can be found [here](https://github.com/openwrt/openwrt/pull/11602#discussion_r1057742078)

Submitted patches can be found here:
- https://patchwork.kernel.org/project/spi-devel-general/patch/20221227141011.111410-2-vincent@vtremblay.dev/
- https://patchwork.kernel.org/project/spi-devel-general/patch/20221227023550.569547-2-vincent@vtremblay.dev/

### Notes

The patches in this PR are adapted versions of the ones submitted upstream to make them compatible with linux v5.15 and v5.10.
The upstream patches are based on v6.2 and the latest spidev driver contains too many changes to be easily backported to v5.15.

Signed-off-by: Vincent Tremblay <vincent@vtremblay.dev>

### CODE REVIEW

File: target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch [hunk 1]
```diff
@@ -1,18 +0,0 @@
-From: Giuseppe Lippolis <giu.lippolis@gmail.com>
```
Comment (BKPepe) [inline target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: I wonder why this patch was removed? I see that the same one was removed from pending-5.15, but there was superseeded by two upstream patches.

Comment (vtremblay) [reply target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: This patch is replaced by the two from this PR to ensure that the manufacturer name is matching the one already used upstream.
I did not add them for 5.10 as I have been told in [another discussion](https://github.com/openwrt/openwrt/pull/11602#discussion_r1056620030) that this is not relevant anymore. I can add them if needed, do you know if 5.10 is still being used on master?

Comment (robimarko) [reply target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: 5.10 is not relevant for IPQ40xx that you were targeting, but if its in generic then 5.10 matters as there are targets still using it

Comment (vtremblay) [reply target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: Ok, I was not aware that it was still being used. I will add the patch for 5.10 as well. Sorry for the confusion!

Comment (robimarko) [reply target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: No worries

Comment (vtremblay) [reply target/linux/generic/pending-5.10/130-add-linux-spidev-compatible-si3210.patch:1]: fixed

**Comment by vtremblay** [comment]
The check for bcm27xx was failing. I have refreshed the patches and force pushed them to fix it.

**Pull request merged**